app/src/main/java/com/imdb_compose/MainActivity.kt

```diff
@@ -1,47 +1,1195 @@
+                modifier = Modifier
+                    .fillMaxWidth(fraction = 0.95f)
+                    .fillMaxHeight(fraction = 0.95f)
+                    .shadow(
+                        color = gray400,
+                        offsetX = 8.dp,
+                        offsetY = 8.dp,
+                        blurRadius = 4.dp,
+                        blurRadiusFilter = "SOLID"
+                    )
+            ) {
+                Column (modifier = Modifier.padding(start = 8.dp, top = 8.dp)) {
+                    Row (
+                        modifier = Modifier.fillMaxWidth(),
+                        horizontalArrangement = Arrangement.SpaceBetween,
+                        verticalAlignment = Alignment.CenterVertically
+                    ) {
+                        // Orange Accent <Catagory>
+                        Row (verticalAlignment = Alignment.CenterVertically) {
+                            Box(
+                                modifier = Modifier
+                                    .padding(start = 8.dp)
+                                    .clip(RoundedCornerShape(24.dp))
+                                    .height(36.dp)
+                                    .width(6.dp)
+                                    .background(ripeMango)
+                            )
+                            Text(
+                                text = catagory,
+                                modifier = Modifier.padding(start = 8.dp),
+                                color = MaterialTheme.colorScheme.onSecondary,
+                                fontSize = MaterialTheme.typography.headlineMedium.fontSize,
+                                fontStyle = MaterialTheme.typography.headlineMedium.fontStyle,
+                                fontWeight = MaterialTheme.typography.headlineMedium.fontWeight
+                            )
+                        }
+                        // See All Btn
+                        Row (verticalAlignment = Alignment.CenterVertically) {
+                            TextButton(
+                                modifier = Modifier.padding(start = 8.dp),
+                                onClick = { navController.navigate(Navigator.CategoryPage(catagory = catagory)) }
+                            ) {
+                                Text(
+                                    text = "See all",
+                                    modifier = Modifier.padding(start = 8.dp),
+                                    color = blue400,
+                                    fontSize = MaterialTheme.typography.headlineSmall.fontSize,
+                                    fontStyle = MaterialTheme.typography.headlineSmall.fontStyle,
+                                    fontWeight = FontWeight.Bold
+                                )
+                            }
+                        }
+                    }
+
+                    Spacer(modifier = Modifier.height(8.dp))
+
+                    when(catagory) {
```

Comment:
I'm still not a fan of this kind of behavior. I think you should just hardcode the order. I could understand when debugging this could be useful, but now that the app is pretty mature I don't think a user would really want the rows to jump around every time they open the app